Contribute/dotnet-style-guide.md

```diff
@@ -0,0 +1,263 @@
+
+## Basic Markdown, GFM, and special characters
+
+You can learn the basics of Markdown, GitHub Flavored Markdown (GFM), and OPS specific extensions in the general articles on [Markdown](how-to-write-use-markdown.md) and [Markdown reference](markdown-reference.md).
+
+Markdown uses special characters such as \*, \`, and \# for formatting. If you wish to include one of these characters in your content, you must do one of two things:
+
+- Put a backslash before the special character to "escape" it (for example, `\*` for a \*)
+- Use the [HTML entity code](http://www.ascii.cl/htmlcodes.htm) for the character (for example, `&#42;` for a &#42;).
+
+## File names
+
+File names use the following rules:
+
+* Contain only lowercase letters, numbers, and hyphens.
+* No spaces or punctuation characters. Use the hyphens to separate words and numbers in the file name.
+* Use action verbs that are specific, such as develop, buy, build, troubleshoot. No -ing words.
+* No small words - don't include a, and, the, in, or, etc.
+* Must be in Markdown and use the .md file extension.
+* Keep file names reasonably short. They are part of the URL for your articles.
+
+## Headings
+
+Use sentence-style capitalization. Always capitalize:
+
+- The first word of a heading.
+- The word following a colon in a title or heading (for example, "How to: Sort an array").
+
+## Text styling
+
+*Italics*
+ Use for files, folders, paths (for long items, split onto their own line), new terms.
+
+**Bold**
+Use for UI elements.
+
+`Code`
+Use for inline code, language keywords, NuGet package names, command-line commands, database table and column names, and URLs that you don't want to be clickable.
+
+## Links
+
+See the general article on [Links](how-to-write-links.md) for information about anchors, internal links, links to other documents, code includes, and external links.
+
+The .NET Docs team uses the following conventions:
+
+- In most cases, we use the relative links and discourage the use of `~/` in links because relative links resolve in the source on GitHub. However, whenever we link to a file in a dependent repo, we'll use the `~/` character to provide the path. Because the files in the dependent repo are in a different location in GitHub the links won't resolve correctly with relative links regardless of how they were written.
+- The C# language specification and the Visual Basic language specification are included in the .NET docs by including the source from the language repositories. The markdown sources are managed in the [csharplang](https://github.com/dotnet/csharplang) and [vblang](https://github.com/dotnet/vblang) repositories.
+
+Links to the spec must point to the source directories where those specs are included. For C#, it's **~/_csharplang/spec** and for VB, it's **~/_vblang/spec**.
+
+- Example: `[C# Query Expressions](~/_csharplang/spec/expressions.md#query-expressions)`
+
+### Links to APIs
```

Comment:
While `xref` links are explained, the description of `cref` links is missing. Often, when one updates the API docs, one updates not the Markdown part, but xml documentation part, where XML documentation rules apply:
https://docs.microsoft.com/en-us/dotnet/csharp/codedoc#see

Reply:
Deferred to #141